Sources/Rendering/WebGPU/RenderWindow/index.js

```diff
@@ -244,19 +244,96 @@ function vtkWebGPURenderWindow(publicAPI, model) {
+          if (model.resizedScreenshotReplacement) {
+            // resize the main canvas back to its original size and show it
+            model.size[0] = model.originalSize[0];
+            model.size[1] = model.originalSize[1];
+
+            // process the resize
+            publicAPI.modified();
+            // restore the saved camera parameters
+            model.renderable.getRenderers().forEach((ren) => {
+              const camera = ren.getActiveCamera();
+              const params = model.activeCamParamMap.get(ren);
+              camera.setFocalPoint(...params.focalPoint);
+              camera.setPosition(...params.position);
+              camera.setParallelScale(params.pScale);
+            });
+            // re-render
+            publicAPI.traverseAllPasses();
+
+            // Clean up the stand-in
+            model.el.removeChild(model.resizedScreenshotReplacement);
+            model.resizedScreenshotReplacement.remove();
+            model.resizedScreenshotReplacement = null;
+            model.activeCamParamMap = null;
+            model.originalSize = null;
+          }
+          resolve(imageURL);
+        } else {
+          // Create a stand-in image overlay while we resize and render
+          const tmpImg = document.createElement('img');
+          tmpImg.width = model.size[0];
+          tmpImg.height = model.size[1];
+          tmpImg.style = {
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            width: '100%',
+            height: '100%',
+          };
+          tmpImg.src = imageURL;
+          model.resizedScreenshotReplacement = model.el.appendChild(tmpImg);
+
+          // save the dimensions of the main canvas
+          model.originalSize = [model.size[0], model.size[1]];
+
+          // hide it
+          model.canvas.style.display = 'none';
+
+          // resize it
+          model.size[0] = model.customCaptureSize[0];
+          model.size[1] = model.customCaptureSize[1];
+          model.customCaptureSize = null;
+
+          // process the resize
+          publicAPI.modified();
+          // save camera parameters before resetCamera so we can restore
+          // them later
+          model.activeCamParamMap = new WeakMap();
+          model.renderable.getRenderers().forEach((ren) => {
+            const camera = ren.getActiveCamera();
+            const fp = camera.getFocalPoint();
+            const pos = camera.getPosition();
+            const pscale = camera.getParallelScale();
+            model.activeCamParamMap.set(ren, {
+              focalPoint: [fp[0], fp[1], fp[2]],
+              position: [pos[0], pos[1], pos[2]],
+              pScale: pscale,
+            });
+            // Without this resetCamera(), we can end up showing only a subset
+            // of the prop(s) depending on the difference between the aspect
+            // ratios of the original size and the custom capture size.
+            ren.resetCamera();
```

Comment:
This should be optional and off by default.

Reply:
Yeah, I was wondering about that.  It's kind of best effort anyway.  To really allow you to get what you want in your screenshot, we'd probably have to call you back before we render and let you compute/set what you want.  

But you think if we just make it optional, the way it is now could be good enough?